adobe/aio-cli-plugin-app-dev
# feat: ACNA-2785 - generate VS Code config for an App Builder Project

## Description:

To debug, you will need to use Visual Studio Code, but you will not have the launch configuration yet in your project. Thus we need a new command to bootstrap your project: `aio app dev init`. 

`aio app dev` debugging does not work without it being in a VS Code context, so we restrict it to that.

1. ~Introduces a new `aio app dev init` command - will install the .vscode/launch.json configuration so users can debug in Visual Studio Code (perhaps this can be run after `aio app init`?)~
2. ~Modifies `aio app dev` so it can only run in the Visual Studio Code context, and not in the regular terminal~

## How Has This Been Tested?

- local link of the plugin
- debug in VS Code of an AB project after running `aio app dev init`
- ~run `aio app dev` in the Terminal (should error out)~
- run `aio app dev init` in the Terminal (should install the .vscode launch config)
- unit tests are expected to fail since we haven't completed unit tests yet

### Test Steps
1. Run the `App Builder: debug actions` configuration in the `VS Code Debug` tab. Modify any `action code`, put a breakpoint in it. Note that the breakpoint will not be "active" until the action is actually hit by calling the action url, only after it is called will the script be loaded, and the execution will stop at the breakpoint
2. Run the `App Builder: debug full stack` configuration in the `VS Code Debug` tab. Modify any `static asset code` (javascript):
    1. Under the `VS Code Debug` tab, find the `Loaded Scripts` section. 
    2. Under that section, find the sub-section `Browser Debug: <your app name>`. 
    3. In there, you will find `__parcel_source_root`
    4. In a sub-heading you will find the static asset source. You can put your breakpoints in the code there.

### Dev Experience Improvements (Future?)

1. Ability to have active breakpoints in the `action` code
5. Ability to set breakpoints in the `web-src` code directly instead of the way outlined above

## Types of changes




## Checklist:
